src/main/java/de/tum/cit/aet/artemis/communication/repository/SavedPostRepository.java

```diff
@@ -80,8 +80,16 @@ public interface SavedPostRepository extends ArtemisJpaRepository<SavedPost, Lon
      *
      * @return List of saved posts of the given user, filtered by the given status.
      */
+    @Query("""
+            SELECT new SavedPost(sp.user, sp.postId, sp.postType, MAX(sp.status), MAX(sp.completedAt))
+            FROM SavedPost sp
+            WHERE sp.user.id = :userId
+                AND sp.status = :status
+            GROUP BY sp.user, sp.postId, sp.postType
+            ORDER BY MAX(sp.completedAt) DESC, MAX(sp.id) DESC
+            """)
     @Cacheable(key = "'saved_post_status_' + #status.getDatabaseKey() + '_' + #userId")
-    List<SavedPost> findSavedPostsByUserIdAndStatusOrderByCompletedAtDescIdDesc(Long userId, SavedPostStatus status);
+    List<SavedPost> findSavedPostsByUserIdAndStatusOrderByCompletedAtDescIdDesc(@Param("userId") Long userId, @Param("status") SavedPostStatus status);
```

Comment:
```suggestion
    List<SavedPost> findSavedPostsByUserIdAndStatusOrderByCompletedAtDescIdDesc(@Param("userId") long userId, @Param("status") SavedPostStatus status);
```
To me it doesn't make sense to call this method with the id being null, so I'd opt for a primitive long rather than a nullable one.

Reply:
Hi! You are right! It doesn't make sense to have a nullable here. I changed it in this [commit](https://github.com/ls1intum/Artemis/pull/10388/commits/e687ffa515c88ccf5a7cfbbdce702b48cdb1d7e6). 👍